### ISSUE TITLE
[Bug] <Enter> does not always add new command if no selection in Object editor

### ISSUE THREAD

**Issue by hsandt** [RPG-Paper-Maker/RPG-Paper-Maker#47]
When there is no selection, <Enter> will not crash but it won't always try to add a new command.

More exactly, pressing <Enter> will do nothing unless the last selected node was a None (empty) command at the end of a block (root or sub-block like If condition).

The issue lies in `WidgetTreeCommands::keyPressEvent` which doesn't check if methods like `contextEdit` actually consume the event before returning.
In addition, actions are not `enabled` exactly when they should (it seems that clicking on a None node enables the New command action and disables the Edit command action, but clicking in the empty area after selecting a true node does not, although it clears the selection).

Fixing one of these two issues will fix this bug. I recommend fixing both to be sure.

1. Make actions return a success/consume bool. Check consumption in `keyPressEvent` and if the event is not consumed, do not return and continue checking the other actions by priority.
2. Make sure you disable the Edit command when selection is cleared, enabling the New command at the same time.

**Comment by hsandt** [comment]
This issue has spawned from #25 to offer a better description of the problem and possible solutions. #25 can now be closed.

**Comment by hsandt** [comment]
This is a valid design choice, as long as we make it clear.

- When clicking in the empty space (outside any entry), the selection highlight becomes semi-transparent. Should it not change at all, if the state is not changing?
- Currently we start with no selection, which should allow us to enter a new entry with <Enter>. If the 1st entry is selected instead, this changes the current behavior, which is OK if you think adding a new entry is immediately is not the most common operation.
- I'm OK to select the command below. I've tried File browser Nautilus and it removes highlight entirely, disabling Enter folder / Open file on <Enter>. However, an invisible selection index seems to remain, as pressing a directional arrow selects a neighbor from the old selected position. But in our case, it's probably better to maintain the selection on the new entry at the selected index.

Actually, having <Enter> mapped to both Edit and New is maybe too disturbing for the user. <Enter> should probably be consistent and only Edit the current selection. However, if the current selection is None, then Enter tries to "edit" None but since there's nothing, it adds a new command instead. If the current selection is not editable like "Else", it does nothing.

Maybe this is actually how you thought it first, and you coded it this way because you considered that None <=> no selection? Anyway, if we always have a selection, what I suggested above happens naturally. We just have to change selection index update a bit.

I haven't thought about pages. I will have a look at it.
